### DISCUSSION TITLE
How to center caption with image (Revealjs slides)

### DISCUSSION THREAD

**Discussion by eteitelbaum** [quarto-dev/quarto-cli#2304] [Q&A] [answered]
I am trying to figure out how to center figure captions under a centered image in Revealjs (although I suppose this would apply to any .html document). Here is my code: 

```
## Growth Under Dictatorships

![Mobutu Sese Seko](images/mobutu.jpg){fig-align="center" width="30%"}

-   How do we explain variation in growth rates under repressive
    regimes?
-   What are the differences between the DRC and other dictatorships
    (e.g. USSR, China)?
```

When I view it in the visual editor, it looks like this: 

![mobutu_snip](https://user-images.githubusercontent.com/62634082/188480489-09a16095-8f89-443d-a039-cb746a8bf12b.PNG)

And it when I have rendered this with R Markdown in the past as a Beamer slide, the caption has also been centered along with the image.

But when I render it as Revealjs the caption gets left awkwardly to side of the image: 

![mobutu_snip2](https://user-images.githubusercontent.com/62634082/188481143-98ceaa45-dc91-4f9c-a470-c446a3599192.PNG)

Is there a way to fix it so that the caption gets centered along with the image?

**Comment by mcanouil** [comment]
You need to provide with a full minimal and reproducible example, especially since you seem to use a template for Reveal.js.

Why are you using `fig-align="center"` since this is the default for captioned figures? See <https://quarto.org/docs/authoring/figures.html#figure-alignment>.
This could be the reason of you output and possibly a bug when default is used explicitly.


_This discussion is related to previous discussions/issues, such as #233, #799, and #1477._

_Edit: also related to #949_

**Comment by eteitelbaum** [comment, marked as answer]
@mcanouil Thanks for linking to those issues. Based on those discussions, it seems like this behavior is not a bug but rather that figure captions are left-justified by design. Further, it seems that captioned figures are centered by default but uncaptioned figures are left justified. 

As you can see, it looks pretty ridiculous when you have a smaller figure and a left-justified caption. I suppose I will have to find some workaround like not using a caption or using a title instead.

**Reply by mcanouil** [reply]
You can embed the whole thing in a div cell:
<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/188497172-258073d4-6223-4e0c-a19b-ece0fa2396d4.png">

<detail><summary>See the code.</summary>

```
---
title: "Untitled"
format: revealjs
---

## Title

::: {style="text-align:center;"}
![Some image alt text](https://quarto.org/quarto.png)
:::
```

</details>

**Reply by mcanouil** [reply]
@eteitelbaum I opened an issue, because there is something a bit fishy now with captioned figures, even for `html` format (see #2306).
In fact, it's more a documentation issue, i.e., old images showing obsolete behaviour.

**Comment by sorenwacker** [comment]
Would be nice to have a feature shortcut `~[My caption](image.png){align=center}` or something along this lines.
